src/connect_options.cpp

```diff
@@ -38,13 +38,13 @@ connect_options::connect_options(const std::string& userName, const std::string&
 void connect_options::set_password(const std::string& password)
 {
 	password_ = password;
-	opts_.password = password_.c_str();
+	opts_.password = password_.empty() ? nullptr : password_.c_str();
```

Comment:
This seems to be the same problem as the pull request #20. The Paho MQTT C requires a `nullptr` as an empty string.